### PR TITLE
proot: 20190510 -> 5.2.0

### PR DESCRIPTION
**Pull request by makefu**
###### Motivation for this change

one month ago proot saw its first release for almost 6.5 years. The official release however needs a certain
pull request to correctly build with python extensions

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/proot/default.nix [hunk 1]
```diff
@@ -19,8 +24,13 @@ stdenv.mkDerivation {
...
 
-  buildInputs = [ talloc ] ++ lib.optional enablePython python;
-  nativeBuildInputs = [ docutils ] ++ lib.optional enablePython swig;
+  buildInputs = [ ncurses libarchive talloc ] ++ lib.optional enablePython python3;
+  nativeBuildInputs = [ pkg-config docutils ] ++ lib.optional enablePython swig;
+  patches = [
+    # without this patch the package does not build with python>3.7
+    (fetchpatch { url = "https://github.com/proot-me/proot/pull/285.patch";
+                  sha256= "1vncq36pr4v0h63fijga6zrwlsb0vb4pj25zxf1ni15ndxv63pxj"; })
+];
```
Comment (veprbl) [inline pkgs/tools/system/proot/default.nix:33]: Can we do a more conventional indentation
```suggestion
    (fetchpatch {
      url = "https://github.com/proot-me/proot/pull/285.patch";
      sha256= "1vncq36pr4v0h63fijga6zrwlsb0vb4pj25zxf1ni15ndxv63pxj";
    })
  ];
```

**Comment by makefu** [comment]
@veprbl : i applied your changes

File: pkgs/tools/system/proot/default.nix [hunk 2]
```diff
@@ -19,8 +24,15 @@ stdenv.mkDerivation {
...
 
-  buildInputs = [ talloc ] ++ lib.optional enablePython python;
-  nativeBuildInputs = [ docutils ] ++ lib.optional enablePython swig;
+  buildInputs = [ ncurses libarchive talloc ] ++ lib.optional enablePython python3;
+  nativeBuildInputs = [ pkg-config docutils ] ++ lib.optional enablePython swig;
+  patches = [
+    # without this patch the package does not build with python>3.7
+    (fetchpatch {
+      url = "https://github.com/proot-me/proot/pull/285.patch";
+      sha256= "1vncq36pr4v0h63fijga6zrwlsb0vb4pj25zxf1ni15ndxv63pxj";
```
Comment (SuperSandro2000) [inline pkgs/tools/system/proot/default.nix:33]:
```suggestion
      url = "https://patch-diff.githubusercontent.com/raw/proot-me/proot/pull/285.patch";
      sha256 = "1vncq36pr4v0h63fijga6zrwlsb0vb4pj25zxf1ni15ndxv63pxj";
```

Comment (veprbl) [reply pkgs/tools/system/proot/default.nix:33]: @SuperSandro2000 
What is the difference?

Comment (makefu) [reply pkgs/tools/system/proot/default.nix:33]: i have the same question. it does not help at all with the readability

Comment (SuperSandro2000) [reply pkgs/tools/system/proot/default.nix:33]: > What is the difference?

The URL redirects to the one I suggested. If you open the patch in your browser you will always end up on the second one. You don't know what GitHub does with the redirect in the future. Also PRs shouldn't be fetched.

Comment (veprbl) [reply pkgs/tools/system/proot/default.nix:33]: I would apply this, if not for the URL, but for `s/sha256=/sha256 =/`

**Pull request merged**